far/dirinfo.cpp

```diff
@@ -111,40 +111,44 @@ int GetDirInfo(string_view const DirName, DirInfoData& Data, multifilter* Filter
 	// Временные хранилища имён каталогов
 	string strLastDirName;
 	os::fs::find_data FindData;
+
+	// Mantis#0002692
+	struct SuspendMacros
+	{
+		SuspendMacros() { Global->CtrlObject->Macro.SuspendMacros(true); }
+		~SuspendMacros() { Global->CtrlObject->Macro.SuspendMacros(false); }
+	} Suspend;
+
 	while (ScTree.GetNextName(FindData,strFullName))
 	{
-		// Mantis#0002692
-		if (!Global->CtrlObject->Macro.IsExecuting())
```

Comment:
As far as I understand, this has been already attempted in f7ae5b23d1864bcb9e22c33112507a4e6a199b23 and reverted in 2f8d7e7a3585b41417b072990a7423bf746c3139 due to issues, discussed in [M#2692](https://bugs.farmanager.com/view.php?id=2692).
Are we positive that suppressing macros fixes those issues?

Reply:
You can test it yourself.
1. Place cursor at some pretty large directory
2. Execute `lua:Keys"EnOut F3 F1"`
3. Wait a couple of seconds, then press `Esc´
   Note 2 things:
   - Operation is successfully interrupted
   - Help screen appears only after operation is finished (or interrupted), and not immediately, like was in f7ae5b23d1864bcb9e22c33112507a4e6a199b23

Reply:
@alabuzhev